core/src/main/java/bio/terra/pearl/core/model/audit/TraceableEntity.java

```diff
@@ -0,0 +1,29 @@
+package bio.terra.pearl.core.model.audit;
+
+import bio.terra.pearl.core.model.BaseEntity;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.SuperBuilder;
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@SuperBuilder
+@NoArgsConstructor
+public class TraceableEntity extends BaseEntity {
+    private UUID responsibleUserId; // id of the user making the change, if it was a participant
+    private UUID responsibleAdminUserId; // id of the user making the change, if it was an admin
+    private String systemProcess; // if the change was the result of an automatic process, store Class+method here
+    private Boolean anonymousUser; // if the change was made by an anonymous user
+    private String justification; // if an admin changes a participant's data, a justification is needed
```

Comment:
I wanted to add some logic in here to enforce that a justification was always provided when a responsibleAdminUserId was provided, but unfortunately we also use this for tracking mailing list contact uploads from the admin tool- which don't need a justification